### DISCUSSION TITLE
[Q] How to use create workbookReader from buffer instead of file path.

### DISCUSSION THREAD

**Discussion by sahilq** [exceljs/exceljs#2375] [Q&A]
I am trying this code to use async row loop

```
const workbookReader = new ExcelJS.stream.xlsx.WorkbookReader(req.file.buffer);
for await (const worksheetReader of workbookReader) {
        for await (const row of worksheetReader) {
          console.log(row);
        }
      }
```

Error: Random characters/Invalid input

**Comment by viniciusls** [comment]
Hi!

Could you figure out what's wrong? I'm facing the same issue (or almost the same). I'm downloading a Buffer from S3 and trying to load it using exceljs.

Thanks in advance!
